bowbuffer.go

```diff
@@ -15,6 +16,168 @@ type Buffer struct {
 	nullBitmapBytes []byte
 }
 
+// NewBuffer returns a new Buffer of size `size` and Type `typ`.
```

Comment:
What justifies this new Buffer implementation ?

Reply:
This change is just moving this function from the genius generated file `bowbuffer.gen.go` to this file. The implementation doesn't change at all here. It will change in the next PR though.